src/modules/powerrename/lib/Settings.cpp

```diff
@@ -3,20 +3,21 @@
 #include <commctrl.h>
 #include "Settings.h"
 #include "PowerRenameInterfaces.h"
-
-const wchar_t c_rootRegPath[] = L"Software\\Microsoft\\PowerRename";
-const wchar_t c_mruSearchRegPath[] = L"SearchMRU";
-const wchar_t c_mruReplaceRegPath[] = L"ReplaceMRU";
-
-const wchar_t c_enabled[] = L"Enabled";
-const wchar_t c_showIconOnMenu[] = L"ShowIcon";
-const wchar_t c_extendedContextMenuOnly[] = L"ExtendedContextMenuOnly";
-const wchar_t c_persistState[] = L"PersistState";
-const wchar_t c_maxMRUSize[] = L"MaxMRUSize";
-const wchar_t c_flags[] = L"Flags";
-const wchar_t c_searchText[] = L"SearchText";
-const wchar_t c_replaceText[] = L"ReplaceText";
-const wchar_t c_mruEnabled[] = L"MRUEnabled";
+#include "resource.h"
+#include <common.h>
+
+const wchar_t* c_rootRegPath = GET_RESOURCE_STRING(IDS_ROOT_PATH).c_str();

```

Comment:
You're creating a temporary `std::wstring` object for each `GET_RESOURCE_STRING` invocation, getting its internal pointer `c_str()` and then the object is immediately destroyed. That means, all these variable will point to a freed memory, possibly leading to a crash and other nasty stuff. Please change the type of these variables to `const std::wstring` and only invoke `.c_str()` where the variable is actually used.

Reply:
I'm not sure we should put those in the resource file anyway. Those are not strings we will localize, those are registry keys PowerRename uses to store its settings.

Reply:
@bzoz is correct, none of these strings should be localized, they are used as IDs and not shown in the UI.

Reply:
@alekhyareddy28 feel free to add a comment in the file, saying that all string in Settings.cpp are used as IDs and not shown in the UI.

Reply:
Sure! I shall add a comment and leave it as is.